app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -105,6 +140,58 @@ private void setUpToolbar() {
         }
     }
 
+    @Subscribe
+    public void tracksDownloadDone(TracksDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new TracksFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void EventsDownloadDone(EventDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new SponsorsFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void sponsorsDownloadDone(SponsorDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new SponsorsFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void microlocationsDownloadDone(TracksDownloadEvent event) {
```

Comment:
Yes I didn't think of this scenario. Will make different events for all the failed events as well or better yet I can add a bool argument to the existing events and set the argument accordingly in case of a success or a failure. Will this be okay?